# 一. 宏观择时策略框架

在研究资产配置的过程中，分析宏观经济基本面是一项重要内容。宏观经济状态与资产收益和风险有着密切的关系，如在经济周期的不同阶段，资产表现出不同的收益和风险特征，分析资产在经济周期不同阶段的表现，有利于管理人分析未来一段时间内资产所处的状态，为资产配置提供有价值的参考依据。
目前，关于经济周期研究已经形成完善的理论框架，按照时间的长短，经济周期又可以分为短周期、中周期、中长周期和长周期。

- 短周期又叫基钦周期，也叫存货周期，将经济周期波动的原因归结为存货投资的变动，短周期一般 4－5 年。
- 中周期又叫朱格拉底周期，将经济的周期变化归结为设备投资的变化，中周期一般 8－10 年。
- 中长周期又叫库兹涅茨周期，一般 18－20年，主要与基础设施建设、产业结构升级和技术进步有关。
- 长周期也叫康德拉杰耶夫周期，平均为 50－60 年。长短经济周期之间存在相互联系和相互作用，当长周期和短周期相互重叠时经济的波动更加明显和剧烈。

经济周期可以划分为上升和下降 2个阶段。上升阶段也称为繁荣阶段，繁荣阶段的最高点称为经济周期的顶峰，下降阶段也称为衰退阶段，衰退阶段的最低点称为经济周期的谷底。
经济周期也可以划分为 4 个阶段，分别是复苏、繁荣、衰退和萧条。复苏是经济在潜在水平之下的扩张，繁荣是经济在潜在水平之上的扩张，衰退是经济在潜在水平之上的收缩，而萧条是经济在潜在水平之下的收缩。 4 阶段的划分方法是对 2阶段划分的进一步细分，以经济的潜在增长水平为划分点，把上升阶段细分为复苏和繁荣两个阶段，把下降阶段划分为衰退和萧条 2 个阶段。

## 1.1 宏观环境对资产价格的影响

当经济处于潜在水平之下的扩张阶段时，也就是经济运行到复苏阶段时，各种生产要素包括资本、劳动力、自然资源都还是过剩的，经济复苏提高了现有生产能力的利用效率，企业的边际收益水平开始提高，国家为了促使经济尽快复苏，采取扩张性的财政政策，加大政府支出，这一系列努力使得企业盈利有所改善，每股盈利开始增长。经济复苏阶段，中央银行一般都会采取比较宽松的货币政策，放松信贷，降低利率，为经济整体扩张提供动力，货币流动性开始推动股票的估值水平提高，每股盈利增长的改善和估值水平的提高推动股票价格上涨，所以这个期间股票表现较好。在这个时期，虽然经济活动在增强，但是通货膨胀通常会持续下降一段时间，低利率导致债券的表现也较好。
当经济处于潜在水平之上的扩张阶段时，经济运行到繁荣阶段，经济增长不可能在原有的速度下持续，劳动力、原材料和生产设备的约束逐步显现出来，经济过热引发了通货膨胀压力，为了控制社会的通货膨胀预期，中央银行将通过提高银行存款准备金、控制信贷、提升利率等一系列手段，为过热的经济降温。利率上升导致债券收益率下降，所以债券的表现就很差。利率上升导致上市公司的资金成本上升，原材料和劳动力约束导致市场上原材料、劳动力价格的上涨，提高了企业的生产成本，尽管部分位于产业链上游的资源型企业能够从资源价格上涨中获利，但是对于大多数生产、加工、制造类企业来说，转嫁生产成本的能力十分有限，成本的上升侵蚀了企业利润，造成企业盈利增速下滑，这时股票市场显出疲态。
当经济处于潜在水平之上的收缩阶段时，经济运行到衰退阶段，经济已经从周期顶峰开始回落，通货膨胀的压力仍然继续存在。在经济回落的过程中，由于总需求减少，生产型企业逐步感受到订单减少，而企业生产成本仍然居高不下，这种总需求减少的压力沿着产业链的从中下游向上游传导，导致企业整体的盈利能力下滑。央行为了控制经济的过度增长而采取紧缩的货币政策，紧缩的货币政策导致了股票的估值水平回落，每股盈利增速回落和估值水平下降，导致股票在这个阶段表现最差。在这个阶段，由于投资者的通货膨胀预期有所减弱，债券的收益率开始走出低谷。
经济处在潜在水平之下的收缩阶段时，经济运行到萧条阶段，此时，由于总需求大幅减少，物价水平出现下降，通货膨胀的压力基本消除。由于市场预期政府会通过降息来防止经济大幅回落，刺激债券收益率上升。总需求大幅减少导致公司在这个阶段的盈利水平很差，盈利增长仍然处在低位，但是，由于货币政策仍然宽松，将股票的估值水平维持在较高水平，股票价格处于比较稳定的状态。随着经济触底，市场对经济增长的预期开始好转，企业盈利出现改善的迹象，货币充裕和经济增长的良好预期进一步推动股票上涨，股票在这个阶段取得最好收益。

## 1.2 选取经济指标

上文中已经简要阐述宏观经济周期对资产价格的影响机制，为了进一步探索宏观经济对资产价格的影响以及为资产配置提供有益的指导，首先需要对宏观经济所处阶段进行划分。国内大多数机构在探索宏观择时策略时，考虑的变量包括国内生产总值、货币供应量、价格指数、工业增加值、利率等等。当然，我们构建宏观择时框架也涉及指标的选择，在选择相关指标时，我们考虑如下因素：一是指标的代表性，即能够充分代表经济整体运行的状况；二是指标的影响性，即能够对经济生活的各方面产生充分的影响；三是数据的可得性，即能够通过相关金融数据终端或国家统计数据集方便获得；四是数据的可算性，即能够通过现有的硬件软件设备计算相关数据。基于此，我们用三个重要的宏观经济变量作为划分划分经济状态的依据，这三个宏观经济变量分别是：国内生产总值 GDP、消费者价格指数 CPI，市场基准利率r。
> 国内生产总值GDP：国内生产总值GDP指按市场价格计算的一个国家(或地区)所有常住单位在一定时期内生产活动的最终成果，它能反映一国经济运行状况的整体走势。持续、稳定提升的 GDP 增长表明国民经济运行状态良好，此时，投资者对当前和未来的实体经济有良好预期，公司利润也会稳定增长，资本市场总体上表现良好。与之相反，在 GDP 增速快速回落时，说明宏观经济体系内存在突出矛盾和问题，此时，投资者对经济和市场的前景十分谨慎，企业盈利情况会逐步恶化，资本市场也会出现大幅波动，一些资产价格会迅速回落。
> 消费者物价指数 CPI：消费者物价指数 CPI代表一般物价水平的变化，是通货膨胀的代理变量之一。在通货膨胀的条件下，物价持续大幅上涨会引起资产价格波动，但波动的方向比较复杂，既可能上涨，也可能下跌。一般来讲，在通货膨胀初始阶段，物价开始上涨，企业盈利能力开始上升，推动股票价格上升。而在通货膨胀上升到较高程度时，价格上升会导致企业生产成本增加，一部分企业的盈利能力出现下降。在严重通货膨胀的条件下，市场会预期政府将紧缩银根、严格控制信贷、控制投资的快速上升、提高市场基准利率等一系列行动，导致债券和股票价格下降。
> 市场利率 Rate：市场利率水平是货币政策的直接体现。一般认为，市场利率对股票、债券和货币资产的收益都有影响。在股票债券的基本模型DDM 模型中，市场利率是折现率的重要组成部分，折现率对资产价格变化有直接影响。

## 1.3 划分宏观经济状态
实践过程中，主流机构在构建宏观因子时，通常会考虑多维度因子，打造完善的多因子体系。比如东方证券的“六维打分体系”，包括：宏观面、中观面、资金面、估值面、情绪面和政策面。宏观面关注的是经济周期；中观面关注的是行业和产业链的景气度；资金面关注的是流动性条件；估值面关注的是贵不贵的问题；情绪面关注的是市场的动量；政策面主要关注的是货币、财政、金融监管（如去杠杆、资管新规等）、行业监管、碳排放等等。
但鉴于目前我们所面临的资源约束，数据可获得性以及算力的限制，我们在构建宏观择时框架时仅选定了三因子，即上文已经详细阐述的国内生产总值GDP、消费者物价指数CPI以及市场利率Rate。并进一步，对每个宏观经济变量考虑两种状态，国内生产总值GDP分为扩张和收缩两种状态、消费者物价指数 CPI 分为上升和下降两种状态、市场利率Rate 分为加息和降息两种状态，这样一共得到 8 种宏观状态。
当然，对于宏观经济变量状态的划分，仍可以继续细分，如国内生产总值GDP可以进一步细分为复苏、繁荣、衰退和萧条四个阶段，还可以加入其它宏观经济变量，如同打标签一样对宏观经济状态进行更细化的描述形成更具体的画像，相关内容将在后续工作中不断完善。

## 1.4 确定可投资的大类资产类别
通常情况下，大类资产配置的对象并不涉及大类资产的内部，而是以指数来代表大类资产，指数反应了大类资产的总体特征和变化规律，常见的指数有债券指数、股票指数、房地产指数、商品指数等，这些指数揭示了一定范围内大类资产的市场运行和发展的总体特征。
当然，如果对大类资产进行合理细分，讨论细分类别资产的特征和运行规律，可以改善投资组合的内部结构，提高资产组合的收益。如按照风格来划分，如果大类资产的任何一个细分类别，具有不同于大类资产的收益和风险特征，就可以被看成是该大类资产的一种风格资产。通常的风格划分存在以下几种形式，股票风格资产可以划分为小股本股票、大股本股票、中等股本股票，或者成长股、价值股等；债券风格资产可以划分为公司债、国债、金融债，或者长期债券、中短期债券，又或者信用债券、担保债券等；商品期货可以划分为农产品期货、金属期货、工业品期货等。
回到本次宏观择时策略中，为了简化分析我们仅考虑大类资产的总体特征，拟选取股票指数、债券指数、商品指数、货币指数，而对于各个大类资产的风格不在细分。风格资产的细化将在后续工作中不断完善，通过细化大类资产的风格特征，提高组合收益。

## 1.5 分析大类资产在不同宏观经济状态下的收益和风险特征
前文已经划分宏观经济状态，根据国内生产总值GDP的扩张和收缩，消费者物价指数CPI的上升和下降，市场利率的加息和降息将宏观经济状态划分为8中状态，如下表所示：
![](media/img_2.png)
根据宏观经济数据，确定不同宏观经济状态所对应的的时间区间，随后分析大类资产，即股票指数、债券指数、商品指数以及货币指数，在不同宏观经济状态所对应的时间区间内的收益和风险特征。
进一步分析大类资产在不同宏观经济状态下的表现，比较总结不同宏观经济状态下哪类资产表现较好，哪类资产表现较差，哪类资产波动率较大，即风险较大，哪类资产波动率较小，即风险较小。

## 1.6 确定不同宏观经济状态下的最优资产配置比例
我们以各种宏观状态下资产收益和协方差作为计算最优资产配置的参数，用均值方差模型计算出投资组合的有效边界，求出一定约束条件下大类资产的最优投资比例。

## 1.7 回测宏观择时策略积累收益表现
假设投资者按照宏观状态进行宏观择时，计算投资组合的积累收益，并考察累计收益的业绩表现。

# 二. 宏观择时策略数据验证

## 2.1 获取数据

### 2.1.1 获取GDP数据

In [1]:
'''
获取GDP数据
从总量上来看，GDP具有明显的季节性特征，因此在分析GDP增速时通常采取剔除季节性影响后的数据，如同比、季调同比
'''
import pandas as pd
data_GDP = pd.read_excel('data/MSTAA_data/GDP.xlsx',
                         sheet_name='data_GDP',
                         index_col='时间',
                         parse_dates=True)
data_GDP['当季同比'] = data_GDP['当季同比'] / 100
data_GDP.sample(3)

当季同比
时间               
2016-03-31  0.069
1992-09-30  0.146
2018-06-30  0.069

In [2]:
import altair as alt


source = pd.DataFrame({
    'x': data_GDP.index,
    'y': data_GDP.当季同比
})

chart = alt.Chart(source).mark_trail().encode(
    x='x:T',
    y='y:Q',
    size='y:Q').properties(
    width = 150*6,
    height=60*6
)

chart

alt.Chart(...)

In [3]:
data_GDP_ = data_GDP[data_GDP.index > '2000-1-1']
data_GDP_.describe()

当季同比
count  89.000000
mean    0.086438
std     0.030933
min    -0.069000
25%     0.069000
50%     0.081000
75%     0.102000
max     0.183000

In [4]:
data_GDP_.median()[0]

0.081

自2000年1月1日至今，我国GDP季度同比增长率的50%分位数为0.081000。因此，采取简化处理，以GDP季度同比增长率的50%分位数为分水岭，超过该分位数指标则认为GDP状态为扩张，低于该分位数指标则认定为GDP状态为收缩。
当然，这样的简化处理存在诸多不合理的地方，比如当前我国宏观经济已经进入新发展阶段，正在逐步从投资拉动转变，房地产投资和城投基建投资的增长模式已经成为过去，高质量发展成为新的路径。因此，当前阶段的增长率势必会回落，单纯以8.1%作为扩张和收缩的分水岭，存在不合理性。
在未来的进一步研究中，将从以下几个方面进一步细化，使其更符合实际情况，如：（1）将宏观经济状态分为四个阶段，即复苏、繁荣、衰退和萧条；（2）依据发展阶段不同，将整个宏观经济划分为多个阶段，就各个阶段的宏观经济特征，设定适合特定宏观经济特征的扩张收缩分水岭。
前述研究思路将在后续策略研究中，逐步深入探索，使得模型更为贴合市场实际，具有实际可操作性。

In [6]:
import altair as alt

source = pd.DataFrame({
    'x':data_GDP_.index,
    'y':data_GDP_.当季同比
})


chart = alt.Chart(source).mark_trail().encode(
    x='x:T',
    y='y:Q',
    size='y:Q').properties(
    width = 150*6,
    height=60*6
)

chart

alt.Chart(...)

In [10]:
# 获取扩张的时间区间
expansion = data_GDP_[data_GDP_.当季同比 > data_GDP_.median()[0]].index
expansion

DatetimeIndex(['2000-03-31', '2000-06-30', '2000-09-30', '2001-03-31',
               '2001-06-30', '2002-03-31', '2002-06-30', '2002-09-30',
               '2002-12-31', '2003-03-31', '2003-06-30', '2003-09-30',
               '2003-12-31', '2004-03-31', '2004-06-30', '2004-09-30',
               '2004-12-31', '2005-03-31', '2005-06-30', '2005-09-30',
               '2005-12-31', '2006-03-31', '2006-06-30', '2006-09-30',
               '2006-12-31', '2007-03-31', '2007-06-30', '2007-09-30',
               '2007-12-31', '2008-03-31', '2008-06-30', '2008-09-30',
               '2009-06-30', '2009-09-30', '2009-12-31', '2010-03-31',
               '2010-06-30', '2010-09-30', '2010-12-31', '2011-03-31',
               '2011-06-30', '2011-09-30', '2011-12-31', '2021-03-31'],
              dtype='datetime64[ns]', name='时间', freq=None)

In [11]:
# 获取收缩的时间区间
contraction = data_GDP_[data_GDP_.当季同比 <= data_GDP_.median()[0]].index
contraction

DatetimeIndex(['2000-12-31', '2001-09-30', '2001-12-31', '2008-12-31',
               '2009-03-31', '2012-03-31', '2012-06-30', '2012-09-30',
               '2012-12-31', '2013-03-31', '2013-06-30', '2013-09-30',
               '2013-12-31', '2014-03-31', '2014-06-30', '2014-09-30',
               '2014-12-31', '2015-03-31', '2015-06-30', '2015-09-30',
               '2015-12-31', '2016-03-31', '2016-06-30', '2016-09-30',
               '2016-12-31', '2017-03-31', '2017-06-30', '2017-09-30',
               '2017-12-31', '2018-03-31', '2018-06-30', '2018-09-30',
               '2018-12-31', '2019-03-31', '2019-06-30', '2019-09-30',
               '2019-12-31', '2020-03-31', '2020-06-30', '2020-09-30',
               '2020-12-31', '2021-06-30', '2021-09-30', '2021-12-31',
               '2022-03-31'],
              dtype='datetime64[ns]', name='时间', freq=None)

In [18]:
for timeindex in data_GDP_.index:
    if timeindex in expansion:
        data_GDP_.loc[timeindex, '状态'] = '扩张'
    if timeindex in contraction:
        data_GDP_.loc[timeindex, '状态'] = '收缩'

data_GDP_

当季同比  状态     颜色
时间                          
2000-03-31  0.087  扩张    red
2000-06-30  0.091  扩张    red
2000-09-30  0.088  扩张    red
2000-12-31  0.075  收缩  green
2001-03-31  0.095  扩张    red
...           ...  ..    ...
2021-03-31  0.183  扩张    red
2021-06-30  0.079  收缩  green
2021-09-30  0.049  收缩  green
2021-12-31  0.040  收缩  green
2022-03-31  0.048  收缩  green

[89 rows x 3 columns]

In [19]:
data_GDP_.to_excel('data/MSTAA_data/GDP_states.xlsx')

将2000年1月1日至今的宏观经济状态划分为扩张和收缩两个阶段，详见下表：
![](media/img_3.png)

### 2.1.2 获取CPI数据
在划分2000年1月至今期间的CPI时，我们以 CPI月度环比和同比数据为基础，编制 CPI 指数，在当前的 CPI 比上一期 CPI 高时，就将当前一期的 CPI 确定为上升状态，在当前一期的 CPI 比上一期的 CPI 低时，就将当前一期确认为 CPI下降阶段。

In [2]:
import pandas as pd
data_CPI = pd.read_excel('data/MSTAA_data/CPI.xlsx',
                         sheet_name='data_CPI',
                         index_col='date',
                         parse_dates=True)
data_CPI.CPI = data_CPI / 100

data_CPI.sample(5)

CPI
date              
2017-07-31  0.0141
2014-05-31  0.0248
2010-06-30  0.0295
2014-07-31  0.0229
2008-02-29  0.0874

In [3]:
for date in data_CPI.index:
    if data_CPI.loc[date, 'CPI'] >= 0:
        data_CPI.loc[date, 'states'] = 1
    else:
        data_CPI.loc[date, 'states'] = 0

data_CPI.sample(5)

CPI  states
date                      
2022-01-31  0.0090     1.0
2015-02-28  0.0143     1.0
2006-10-31  0.0141     1.0
2013-11-30  0.0302     1.0
2003-10-31  0.0181     1.0

In [4]:
import altair as alt

source = pd.DataFrame({
    'x':data_CPI.index,
    'y':data_CPI.CPI
})

chart = alt.Chart(source).mark_trail().encode(
    x='x:T',
    y='y:Q',
    size='y:Q').properties(
    width = 150*6,
    height=60*6)

chart

alt.Chart(...)

data_CPI数据集中states列的1表示CPI上升，即通胀，0表示CPI下降，即通缩。

In [28]:
# 存储CPI_states
data_CPI.to_excel('data/MSTAA_data/CPI_states.xlsx')

将2000年1月1日至今的宏观经济状态根据CPI划分为扩张和收缩两个阶段，详见下表：
![](media/img_4.png)

In [1]:
import pandas as pd

data_RATE = pd.read_excel('data/MSTAA_data/rate.xlsx',
                          sheet_name='rate',
                          index_col='date',
                          parse_dates=True,
                          na_filter=True)

data_RATE.fillna(method='ffill', inplace=True)
data_RATE_ = data_RATE.apply(lambda x : x / 100)
data_RATE_.sample(3)

rate
date              
2014-04-11  0.0577
2011-08-04  0.0656
2014-04-04  0.0577

In [2]:
import altair as alt

source = pd.DataFrame({
    'x':data_RATE_.index,
    'y':data_RATE_.rate})

chart = alt.Chart(source).mark_trail().encode(
    x='x:T',
    y='y:Q',
    size='y:Q').properties(
    width = 150*6,
    height=60*6)

chart

alt.Chart(...)

In [3]:
# 保存利率数据
data_RATE_.to_excel('data/MSTAA_data/rate_states.xlsx')

将2000年1月1日至今的宏观经济状态根据rate划分为加息和降息两个阶段，详见下表：
![](media/img_5.png)

根据宏观经济的GDP、CPI及Rate的情况，将宏观经济状态划分为八种状态，详见下表：
![](media/img_6.png)
状态2和状态6出现的天数为0，从经济学角度不难解释，即宏观经济处于通缩的情况下，加息是不可取的，而同理在宏观经济处于通胀的情况下，降息同样不可取。
基于上述状态分类，汇总整理了各个状态对应的时间。

In [4]:
import pandas as pd

states = pd.read_excel('data/MSTAA_data/states.xlsx',
                       sheet_name='states',
                       parse_dates=True)
states.sample(3)

states1  states2    states3 states4 states5  states6    states7  \
1218     NaT      NaN 2014-10-24     NaT     NaT      NaN 2010-06-17   
3094     NaT      NaN 2019-12-13     NaT     NaT      NaN        NaT   
1093     NaT      NaN 2014-06-21     NaT     NaT      NaN 2010-02-12   

        states8  
1218 2007-12-05  
3094        NaT  
1093 2007-08-02

In [5]:
import pandas as pd
states_timeline = pd.read_excel('data/MSTAA_data/states.xlsx',
                                sheet_name='states_timeline',
                                parse_dates=True)
states_timeline.sample(3)

date  state
4350 2002-06-25      5
5765 2004-07-15      7
3039 2019-04-22      3